tox.ini

```diff
@@ -10,97 +15,163 @@ minversion = 4.22.0
 
-[testenv:cov]
+# --- coverage reporting
+[testenv:cov_base]
+description = <base for coverage envs>
 dependency_groups = coverage
 skip_install = true
+
 [testenv:cov-clean]
-base = cov
+base = cov_base
+description = Clear existing coverage data
 commands = coverage erase
+
 [testenv:cov-combine]
-base = cov
+base = cov_base
+description = Combine coverage data from parallel tests
 commands = coverage combine
-depends = py{3.13,3.12,3.11,3.10,3.9}{,-mindeps}
+depends = py{3.13,3.12,3.11,3.10,3.9}{,-mindeps,-localsdk,-sdkmain}
+
 [testenv:cov-report]
-base = cov
+base = cov_base
+description = Report coverage data
 commands_pre = coverage html --fail-under=0
 commands = coverage report
 depends = cov-combine
+
 [testenv:cov-report-ci]
-base = cov
+base = cov_base
+description = Combine and report coverage data (for use in CI)
 commands =
     coverage combine
     coverage report
 
+
+# --- various linters
 [testenv:lint]
+description = Run pre-commit fixers and linters
 deps = pre-commit
 recreate = true
 skip_install = true
 commands = pre-commit run --all-files
 
-[testenv:mypy]
+
+# --- type checking with mypy
+[testenv:mypy_base]
+description = <base for mypy envs>
 dependency_groups = typing
 commands = mypy {posargs:src/}
+
+[testenv:mypy]
+base = mypy_base
+description = Type check with mypy
+
 [testenv:mypy-minpython]
-base = mypy
+base = mypy_base
+description = Type check with mypy for our minimum supported Python
 commands = mypy --python-version "3.9" {posargs:src/}
+
 [testenv:mypy-maxpython]
-base = mypy
+base = mypy_base
+description = Type check with mypy for our maximum supported Python
 commands = mypy --python-version "3.13" {posargs:src/}
 
+
+# --- doc builds
 [testenv:reference]
+description = Generate reference doc for inclusion in docs.globus.org builds
 allowlist_externals = find
 commands_pre = find reference/ -name "*.adoc" -type f -delete
 commands = python ./reference/_generate.py {posargs}
 
-[testenv:check-min-python-is-tested]
-description = Check the Requires-Python metadata against CI config
+
+# --- check package and repo metadata
+[testenv:check_metadata_base]
+description = <base env for package and repo metadata checks>
 skip_install = true
 dependency_groups = check-project-metadata
-commands = python scripts/ensure_min_python_is_tested.py
-
-[testenv:check-package-data{,-minimum}]
-# NOTE: base_python should be the minimum supported Python
+# NOTE: base_python should be the minimum supported Python for `minimum` envs
 # the tools in `check-project-metadata` require Python 3.10+
 # when we drop 3.9 support, this will become correct
-base_python = python3.10
-recreate = true
-skip_install = true
-dependency_groups = check-project-metadata
+base_python =
+    minimum: python3.10
+    !minimum: python
+# these checks build without build isolation, so install the build backend explicitly
```

Comment:
This phrasing appears to suggest a causal relationship that isn't intended.

> _Because_ build isolation is disabled, we _must therefore_ install the build backend ourselves.

Rather, the goal is to test that the package can be built on the minimum Python platform with the minimum version of the build backend, and `build` has no way to control for that, so we've wrested dependency management from `build` and given it to `tox`.

I think that the comment could be removed, or could be moved to the `commands` point where `--no-build-isolation` is passed.

Reply:
Yeah, looks good to me!